### PR TITLE
Jsapi: Implement missing EncoderConfig methods

### PR DESCRIPTION
**Pull request by Urhengulas**
## rav1e_js
* add `EncoderConfig.setColorDescription(...)`
* add `EncoderConfig.setMasteringDisplay(...)`
* `encoder`
    * Pass `EncoderConfig` as ref to constructor
      --> This enables still using the config afterwards
    * Rename [FrameEncoder, VideoEncoder]::fromEncoderConfig(...) to ::new(...)
    * Rm `::default()` from `FrameEncoder`, `VideoEncoder`
* `mod`: `pub use encoder::VideoEncoder`
* `tests`: Clean `utils::simple_encoding`
* bump version to `0.2.1`

### CODE REVIEW

File: src/api/color.rs [hunk 1]
```diff
@@ -41,33 +41,34 @@ pub use v_frame::pixel::ChromaSampling;
...
+  Unspecified = 0,
   /// BT.709
   BT709 = 1,
-  /// Unspecified, must be signaled or inferred outside of the bitstream
-  Unspecified,
-  /// BT.470 System M (historical)
-  BT470M = 4,
   /// BT.470 System B, G (historical)
-  BT470BG,
+  BT470BG = 2,
```
Comment (vibhoothiiaanand) [inline src/api/color.rs:55]: AFAIK these numbers are defined as per spec `ISO/IEC 23091-4/ITU-T H.273`
Here 2 is unspecified, 3 is reserved so you cannot keep numbers like that IIUC.

File: src/api/color.rs [hunk 2]
```diff
@@ -81,45 +82,46 @@ impl Default for ColorPrimaries {
 /// Supported Transfer Characteristics
 ///
 /// As defined by “Transfer characteristics” section of ISO/IEC 23091-4/ITU-TH.273.
+#[wasm_bindgen]
 #[derive(
   ArgEnum, Debug, Clone, Copy, PartialEq, FromPrimitive, Serialize, Deserialize,
 )]
 #[repr(C)]
 pub enum TransferCharacteristics {
+  /// Unspecified, must be signaled or inferred outside of the bitstream
```
Comment (vibhoothiiaanand) [inline src/api/color.rs:91]: Same goes for this also

File: src/api/color.rs [hunk 3]
```diff
@@ -139,31 +142,31 @@ pub enum MatrixCoefficients {
   /// Identity matrix
   Identity = 0,
   /// BT.709
-  BT709,
+  BT709 = 1,
   /// Unspecified, must be signaled or inferred outside of the bitstream.
-  Unspecified,
+  Unspecified = 2,
```
Comment (vibhoothiiaanand) [inline src/api/color.rs:147]: Here again

File: src/api/color.rs (hunk 1 above)
Comment (Urhengulas) [reply src/api/color.rs:55]: Okay. That is basically what my question is about.

The problem is, that I either need to specify no or all options in the enum with a number.
I will have a look into the spec.

File: src/api/color.rs [hunk 4]
```diff
@@ -41,33 +41,34 @@ pub use v_frame::pixel::ChromaSampling;
 /// Supported Color Primaries
 ///
 /// As defined by “Color primaries” section of ISO/IEC 23091-4/ITU-T H.273
+#[wasm_bindgen]
```
Comment (lu-zero) [inline src/api/color.rs:44]: wasm_bindgen should not require this, can you poke upstream and ask them to fix that in the macro?

**Comment by Urhengulas** [comment]
Good idea

File: src/api/color.rs (hunk 4 above)
Comment (Urhengulas) [reply src/api/color.rs:44]: Will do

Comment (Urhengulas) [reply src/api/color.rs:44]: https://github.com/rustwasm/wasm-bindgen/issues/2273

**Comment by lu-zero** [comment]
The version should be `0.3` since you are breaking the API by removing `fromEncoderConfig` and `default`.

File: rav1e_js/src/encoder_config.rs [hunk 5]
```diff
@@ -270,3 +395,109 @@ impl EncoderConfig {
     self.clone()
   }
 }
+
+/// Supported Color Primaries
+///
+/// As defined by “Color primaries” section of ISO/IEC 23091-4/ITU-T H.273
+#[wasm_bindgen]
+pub enum ColorPrimaries {
```
Comment (vibhoothiiaanand) [inline rav1e_js/src/encoder_config.rs:403]: Not sure why are we re-defining here, 
We already have it in public API right, 
https://github.com/xiph/rav1e/blob/425c5ca9ccb08ff64c81f495e7fe109a891cae4f/src/api/color.rs#L48-L73

File: rav1e_js/src/encoder_config.rs [hunk 6]
```diff
@@ -270,3 +395,109 @@ impl EncoderConfig {
...
+  SMPTE432,
+  /// EBU Tech. 3213-E
+  EBU3213,
+}
+
+/// Supported Transfer Characteristics
+///
+/// As defined by “Transfer characteristics” section of ISO/IEC 23091-4/ITU-TH.273.
+#[wasm_bindgen]
+pub enum TransferCharacteristics {
```
Comment (vibhoothiiaanand) [inline rav1e_js/src/encoder_config.rs:434]: Same comment as previous

File: rav1e_js/src/encoder_config.rs [hunk 7]
```diff
@@ -270,3 +395,109 @@ impl EncoderConfig {
...
+  /// SMPTE ST 428
+  SMPTE428,
+  /// BT.2100 HLG (Hybrid Log Gamma), ARIB STD-B67
+  HLG,
+}
+/// Matrix coefficients
+///
+/// As defined by the “Matrix coefficients” section of ISO/IEC 23091-4/ITU-TH.273.
+#[wasm_bindgen]
+pub enum MatrixCoefficients {
```
Comment (vibhoothiiaanand) [inline rav1e_js/src/encoder_config.rs:474]: Ditto as previous

File: rav1e_js/src/encoder_config.rs (hunk 5 above)
Comment (vibhoothiiaanand) [reply rav1e_js/src/encoder_config.rs:403]: https://docs.rs/rav1e/0.3.3/rav1e/color/enum.ColorPrimaries.html

Comment (Urhengulas) [reply rav1e_js/src/encoder_config.rs:403]: It is not possible to export these enums. See https://github.com/rustwasm/wasm-bindgen/issues/2273 for the problem.

Therefore I'm having a copy of them without annotated discriminants as a workaround until it is possible to export them.

File: src/api/color.rs [hunk 8]
```diff
@@ -226,6 +227,16 @@ pub struct ChromaticityPoint {
   pub y: u16,
 }
 
+#[wasm_bindgen]
+impl ChromaticityPoint {
+  /// Chromaticity coordinates as defined by CIE 1931, expressed as 0.16
+  /// fixed-point values.
+  #[wasm_bindgen(constructor)]
+  pub fn constructor(x: u16, y: u16) -> Self {
```
Comment (lu-zero) [inline src/api/color.rs:235]: please add a `#[cfg()]` to make so it is not added to the non-js API, why is that needed btw?

File: rav1e_js/src/encoder_config.rs (hunk 5 above)
Comment (lu-zero) [reply rav1e_js/src/encoder_config.rs:403]: Please add a note about it.

File: src/api/color.rs (hunk 8 above)
Comment (Urhengulas) [reply src/api/color.rs:235]: So the js user can add a new `ChromaticityPoint` via:
```ts
let a = new ChromaticityPoint(0, 0);
```

**Comment by Urhengulas** [comment]
Alex Crichton confirmed that there is currently no better way than our current workaround for the enum-problem (https://github.com/rustwasm/wasm-bindgen/issues/2273#issuecomment-685965892).
So I would propose to actually go with it for now. Even if it is kinda not beautiful...

**Comment by barrbrain** [comment]
If this is still relevant, the PR should be ported to https://github.com/rust-av/rav1e_js
